### PR TITLE
All courses

### PR DESCRIPTION
**Pull request by firepolo**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Fixed issues | comma-separated list of issues fixed by the PR, if any

Add data source for display all courses in a widget.
The goal is to get a list of all courses without the training sessions.

### CODE REVIEW

File: src/plugin/cursus/Listener/DataSource/AllCourseSource.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        $this->finder = $finder;
+    }
+
+    public function getData(GetDataEvent $event)
+    {
+        $options = $event->getOptions();
+
+        $event->setData(
+            $this->finder->search(Course::class, $options)
```
Comment (Elorfin) [inline src/plugin/cursus/Listener/DataSource/AllCourseSource.php:33]: Course must be filtered by organizations. See : https://github.com/claroline/Claroline/blob/13.x/src/plugin/cursus/Controller/CourseController.php#L91.

File: src/plugin/cursus/Listener/DataSource/AllCourseSource.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * file that was distributed with this source code.
+ */
+
+namespace Claroline\CursusBundle\Listener\DataSource;
+
+use Claroline\AppBundle\API\FinderProvider;
+use Claroline\CoreBundle\Event\DataSource\GetDataEvent;
+use Claroline\CursusBundle\Entity\Course;
+
+class AllCourseSource
```
Comment (Elorfin) [inline src/plugin/cursus/Listener/DataSource/AllCourseSource.php:18]: Should be `AllCoursesSource`

File: src/plugin/cursus/Resources/config/config.yml [hunk 3]
```diff
@@ -21,6 +21,11 @@ plugin:
           context: [desktop]
           tags: [community, event]
 
+        - name: all_courses
+          type: list
+          context: [workspace, desktop, home]
```
Comment (Elorfin) [inline src/plugin/cursus/Resources/config/config.yml:26]: It shouldn't be available in `workspace` context.

File: src/plugin/cursus/Resources/modules/data/sources/courses.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+export default {
+  name: 'courses',
+  icon: 'fa fa-fw fa-cubes',
+  parameters: {
+    primaryAction: (course) => ({
+      type: URL_BUTTON,
+      target: `#${route(toolRoute('trainings')+'/catalog', course)}`
+    }),
+    definition: [
```
Comment (Elorfin) [inline src/plugin/cursus/Resources/modules/data/sources/courses.js:17]: Fill the definition with the same columns than : https://github.com/claroline/Claroline/blob/13.x/src/plugin/cursus/Resources/modules/course/components/list.jsx#L31.

**Review by Elorfin — CHANGES_REQUESTED** [review]

File: src/plugin/cursus/Listener/DataSource/AllCourseSource.php (hunk 1 above)
Comment (firepolo) [reply src/plugin/cursus/Listener/DataSource/AllCourseSource.php]: I'm done

File: src/plugin/cursus/Listener/DataSource/AllCourseSource.php (hunk 2 above)
Comment (firepolo) [reply src/plugin/cursus/Listener/DataSource/AllCourseSource.php]: I'm fixed

File: src/plugin/cursus/Resources/config/config.yml (hunk 3 above)
Comment (firepolo) [reply src/plugin/cursus/Resources/config/config.yml]: I'm removed workspace

File: src/plugin/cursus/Resources/modules/data/sources/courses.js (hunk 4 above)
Comment (firepolo) [reply src/plugin/cursus/Resources/modules/data/sources/courses.js:18]: I'm filled definition with all columns

**Review by Elorfin — APPROVED** [review]
👍

**Pull request merged**